code/modules/clothing/clothing.dm

```diff
@@ -74,19 +74,41 @@
 
 /obj/item/clothing/update_icon()
 	..()
-
 	overlays.len = 0
 	dynamic_overlay.len = 0
+	var/image/dyn_overlay_worn
+	var/image/dyn_overlay_left
+	var/image/dyn_overlay_right
+
+	if ((luminosity > 0) || (dyed_parts.len > 0))
```

Comment:
Nit, but relational operators always are checked prior to boolean operators so your parentheses are not necessary.

Reply:
double nit, `dyed_parts.len > 0` could be simplified to `dyed_parts.len`